pkg/platform/packet/packet.go

```diff
@@ -237,6 +255,102 @@ func (c *config) terraformSmartApply(ex *terraform.Executor, dnsProvider dns.DNS
 	return ex.Apply()
 }
 
+// terraformAddDeps adds explicit dependencies to cluster nodes so nodes
+// with a specific hw reservation UUID are created before nodes that don't have
+// a specific hw reservation UUID.
+// The function modifies c.NodesDependsOn and c.workerPools[].NodesDependsOn,
+// assigning to them a string, formatted as a Terraform list, of all worker
+// Terraform targets needed to be created first. For example:
+//
+//	// Suppose worker pool "example" is the only to use a specific hw
+//	// reservation IDs. IOW, the controllers (c.NodesDependsOn) depends on
+//	// worker pool "example" to be created first.
+//	// Then, after calling this function, the attribute will be:
+// 	c.NodesDependsOn = "[ module.worker-example.worker_nodes_ids ]"
```

Comment:
This comment is no longer valid, as we changed the type to slice.

Reply:
Ouch! Sorry, I didn't ask for another review (just pushed some of the changes) because I knew some things might not be accurate and wanted to have a look myself. But thanks for catching it, I'm not sure if I would catch it (I usually need fresh eyes to catch those things :D)

Reply:
Solved this and several other outdated comments :)